db/db_sst_test.cc

```diff
@@ -584,13 +614,84 @@ TEST_F(DBSSTTest, CancellingCompactionsWorks) {
   ASSERT_OK(Flush());
   dbfull()->TEST_WaitForCompact(true);
 
+  // Because we set a callback in CancelledCompaction, we actually
+  // let the compaction run
```

Comment:
In retrospect, we probably should've adapted the old test cases to the new reality (compactions can fail due to space reservation), rather than using the callback to avoid it. Because now we're testing the test workaround and it's getting kinda confusing :p. (I don't suggest addressing this comment in this PR.)